src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -89,9 +160,41 @@ public SocksBeginExBuilder typeId(
         }
 
         public SocksBeginExBuilder address(
-            String address)
+            String address) throws UnknownHostException
         {
-            this.beginExRW.address(address);
+            if (IPV4_ADDRESS_MATCHER.get().reset(address).matches())
+            {
+                final Matcher ipv4Matcher = IPV4_ADDRESS_MATCHER.get();
+                final byte[] ipv4AddressBytes = IPV4_ADDRESS_BYTES.get();
+                for (int group = 1; group < ipv4AddressBytes.length + 1; group++)
+                {
+                    ipv4AddressBytes[group - 1] = parseByte(ipv4Matcher.group(group), 10);
+                }
+                beginExRW.address(b -> b.ipv4Address(s -> s.set(ipv4AddressBytes)));
+            }
+            else if (IPV6_STD_ADDRESS_MATCHER.get().reset(address).matches())
+            {
+                final byte[] addressBytes = IPV6_ADDRESS_BYTES.get();
+                final Matcher ipv6Matcher = IPV6_STD_ADDRESS_MATCHER.get();
+                Arrays.fill(addressBytes, (byte) 0);
+                for (int group = 1; group < ipv6Matcher.groupCount() + 1; group++)
+                {
+                    String groupHex = ipv6Matcher.group(group);
+                    fillInBytes(addressBytes, 2 * (group - 1), groupHex);
+                }
```

Comment:
Hoist lines `179-184` into method called `encodeStandardIpv6AddressBytes` and reuse for `routeEx`.

Reply:
done